rpc/types_executables.go

```diff
@@ -0,0 +1,362 @@
+	AssertLeIsSecondArcExcluded AssertLeIsSecondArcExcluded `json:",omitempty"`
+	DebugPrint                  DebugPrint                  `json:",omitempty"`
+	DivMod                      DivMod                      `json:",omitempty"`
+	EvalCircuit                 EvalCircuit                 `json:",omitempty"`
+	Felt252DictEntryInit        Felt252DictEntryInit        `json:",omitempty"`
+	Felt252DictEntryUpdate      Felt252DictEntryUpdate      `json:",omitempty"`
+	FieldSqrt                   FieldSqrt                   `json:",omitempty"`
+	GetCurrentAccessDelta       GetCurrentAccessDelta       `json:",omitempty"`
+	GetCurrentAccessIndex       GetCurrentAccessIndex       `json:",omitempty"`
+	GetNextDictKey              GetNextDictKey              `json:",omitempty"`
+	GetSegmentArenaIndex        GetSegmentArenaIndex        `json:",omitempty"`
+	InitSquashData              InitSquashData              `json:",omitempty"`
+	LinearSplit                 LinearSplit                 `json:",omitempty"`
+	RandomEcPoint               RandomEcPoint               `json:",omitempty"`
+	ShouldContinueSquashLoop    ShouldContinueSquashLoop    `json:",omitempty"`
+	ShouldSkipSquashLoop        ShouldSkipSquashLoop        `json:",omitempty"`
+	SquareRoot                  SquareRoot                  `json:",omitempty"`
+	TestLessThan                TestLessThan                `json:",omitempty"`
+	TestLessThanOrEqual         TestLessThanOrEqual         `json:",omitempty"`
+	TestLessThanOrEqualAddress  TestLessThanOrEqualAddress  `json:",omitempty"`
+	U256InvModN                 U256InvModN                 `json:",omitempty"`
+	Uint256DivMod               Uint256DivMod               `json:",omitempty"`
+	Uint256SquareRoot           Uint256SquareRoot           `json:",omitempty"`
+	Uint512DivModByUint256      Uint512DivModByUint256      `json:",omitempty"`
+	WideMul128                  WideMul128                  `json:",omitempty"`
+}
+
+type StarknetHint struct {
+	Cheatcode  Cheatcode  `json:",omitempty"`
+	SystemCall SystemCall `json:",omitempty"`
+}
+
+type DeprecatedHintEnum string
+
+const (
+	AssertCurrentAccessIndicesIsEmpty DeprecatedHintEnum = "AssertCurrentAccessIndicesIsEmpty"
+	AssertAllKeysUsed                 DeprecatedHintEnum = "AssertAllKeysUsed"
+	AssertLeAssertThirdArcExcluded    DeprecatedHintEnum = "AssertLeAssertThirdArcExcluded"
+)
+
+type AssertAllAccessesUsed struct {
+	NotUsedAccesses CellRef `json:"n_used_accesses"`
+}
+
+type CellRef struct {
+	Register Register `json:"register"`
+	Offset   int      `json:"offset"`
+}
+
+type Register string
+
+const (
+	AP Register = "AP"
+	FP Register = "FP"
+)
+
+type AssertLtAssertValidInput struct {
+	A ResOperand `json:"a"`
+	B ResOperand `json:"b"`
+}
+
+type felt252Dict struct {
+	DictPtr ResOperand `json:"dict_ptr"`
+	Key     ResOperand `json:"key"`
+}
+
+type Felt252DictRead struct {
+	felt252Dict
+	ValueDst CellRef `json:"value_dst"`
+}
+
```

Comment:
Why seperate ValueDst from DictPtr and Key?

Reply:
Ah, because the read and write both use the same types for dict_ptr and key?

Reply:
Yup, [Felt252DictEntryInit](https://github.com/NethermindEth/starknet.go/pull/642/files#diff-a5d4b8f184d307383cc60d205757b876abb2e7cf9c45a4a49dfb845871fa2910R139) too